var/spack/repos/builtin/packages/msmpi/package.py

```diff
@@ -3,40 +3,54 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
+import platform
+import re
+
+from spack.build_systems.generic import GenericBuilder
 from spack.package import *
 
 
 class Msmpi(Package):
-    """A Windows-specced build of MPICH provided directly by
-    Microsoft Support Team
-    """
+    """MSMPI is a Windows port of MPICH provided by the Windows team"""
 
-    homepage = "https://www.microsoft.com/en-us/download/default.aspx"
-    maintainers = ["jpopelar"]
+    homepage = "https://docs.microsoft.com/en-us/message-passing-interface/microsoft-mpi"
+    url = "https://github.com/microsoft/Microsoft-MPI/archive/refs/tags/v10.1.1.tar.gz"
+    git = "https://github.com/microsoft/Microsoft-MPI.git"
 
     executable = ["mpiexec.exe"]
 
-    version(
-        "10.0",
-        sha256="7dae13797627726f67fab9c1d251aec2df9ecd25939984645ec05748bdffd396",
-        extension="exe",
-        expand=False,
-    )
+    version("10.1.1", sha256="63c7da941fc4ffb05a0f97bd54a67968c71f63389a0d162d3182eabba1beab3d")
+    version("10.0.0", sha256="cfb53cf53c3cf0d4935ab58be13f013a0f7ccb1189109a5b8eea0fcfdcaef8c1")
 
     provides("mpi")
 
-    conflicts("platform=linux")
-    conflicts("platform=darwin")
-    conflicts("platform=cray")
+    depends_on("win-wdk")
 
-    def url_for_version(self, version):
-        return "https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisetup.exe"
+    patch("ifort_compat.patch")
 
-    def determine_version(self, exe):
-        output = Executable("mpiexec.exe")
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)()
         ver_str = re.search("[Version ([0-9.]+)]", output)
         return Version(ver_str.group(0)) if ver_str else None
 
+
+class GenericBuilder(GenericBuilder):
+    def is_64bit(self):
+        return platform.machine().endswith("64")
+
+    def build_command_line(self):
+        arch = "intel64" if self.is_64() else "ia32"
+        args = ["-noLogo"]
+        # The argument may say gfortran, but spack patches MSMPI to be compatible with IFortran
+        # however that variable name proved difficult to patch, so it remians the same
+        args.append("/p:GFORTRAN_BIN=%sbin\\%s" % (os.environ["IFORT_COMPILER21"], arch))
```

Comment:
We specify OneAPI as an optional prereq for running Spack on Windows, and the MSVC compiler setup environment method will ensure this is present (from vcvars env setup script) so this will be present if the compiler has an associated fortran component (msvc compiler class just uses ifort). I'll add a check to ensure the compiler has an associated fortran.